src/jose/JWT.js

```diff
@@ -25,245 +50,363 @@ class JWT extends JSONDocument {
-   * @param {String} data
-   * @param {Object} [options]
+   * @param {Object|String} data
+   * @param {String} [data.serialized] - Existing serialized JWT
    *
-   * @returns {Promise<JWT>}
+   * @return {JWT}
    */
-  static verify (key, data, options={}) {
+  static from (data) {
     let ExtendedJWT = this
-    let token
 
-    try {
-      token = ExtendedJWT.decode(data)
-    } catch (err) {
-      return Promise.reject(err)
+    // Decode serialized token
+    if (typeof data === 'string' || data.serialized) {
+      return this.decode(data.serialized || data)
     }
 
-    let { signatures } = token
+    let { payload, signatures, serialization } = data
 
-    // Assign options to JWT as nonenumerable properties
-    Object.keys(options).forEach(field => {
-      Object.defineProperty(token, field, { value: options[field], enumerable: false })
-    })
+    if (!payload) {
+      throw new DataError('Invalid JWT')
+    }
 
-    // Map keys to signatures by index
-    if (Array.isArray(key) && signatures) {
-      token.signatures = signatures.map((descriptor, index) => {
-        // TODO more sophisticated key mapping using hints like `kid'
-        if (index < key.length) {
-          Object.defineProperty(descriptor, 'key', { value: key[index], enumerable: false })
-        }
+    // Include compelete signature descriptors only
+    if (signatures && Array.isArray(signatures)) {
+      signatures = signatures.filter(descriptor => !descriptor.cryptoKey || descriptor.signature)
+    } else {
+      signatures = []
+    }
 
-        return descriptor
-      })
+    // Normalize existing flat signature
+    if (!data.cryptoKey && data.signature) {
+      let { protected: protectedHeader, header: unprotectedHeader, signature } = data
+      let descriptor = {}
+
+      if (!protectedHeader && unprotectedHeader) {
+        descriptor.protected = unprotectedHeader
+      } else {
+        descriptor.protected = protectedHeader
+        descriptor.header = unprotectedHeader
+      }
+
+      descriptor.signature = signature
 
-    // Assign single key to all signatures as nonenumerable property
-    } else if (signatures) {
-      token.signatures = signatures.map(descriptor => {
-        Object.defineProperty(descriptor, 'key', { value: key, enumerable: false })
-        return descriptor
+      if (signatures && Array.isArray(signatures)) {
+        signatures.unshift(descriptor)
+      } else {
+        signatures = [descriptor]
+      }
+    }
+
+    return new ExtendedJWT(
+      clean({
+        payload,
+        signatures,
+        serialization,
+        type: 'JWS'
       })
+    )
+  }
 
-    // Assign key to token as nonenumerable property
-    } else {
-      Object.defineProperty(token, 'key', { value: key, enumerable: false })
+  /**
+   * sign
+   *
+   * @description
+   * Sign a JSON Web Token
+   *
+   * @params {...Object} data - Token data
+   *
+   * @returns {Promise<SerializedToken>}
+   */
+  static sign (...data) {
```

Comment:
Something about this approach doesn't feel right. 
In the usage examples in RS256-JWT-Sign.js, all of those would work just as well if all those properties were within a single object.

I see the intent with the myTokenData example, but seems to be a fairly rare use case, and not really idiomatic to Javascript. I'm not sure it's worth the potential developer confusion.

We're already future-proofing the API in other places, by just having a single flexible options object.